pkg/test/context.go

```diff
@@ -41,43 +37,57 @@ type TestCtx struct { //nolint:golint
 	skipCleanupOnError bool
 }
 
+// todo(camilamacedo86): Remove the following line just added for we are able to deprecated TestCtx
+// need to be done before: 1.0.0
+
+// Deprecated: Context exists for historical compatibility
+// Yoou should use Context instead of.
+type TestCtx = Context //nolint:golint
+
 type CleanupOptions struct {
-	TestContext   *TestCtx
+	TestContext   *Context
 	Timeout       time.Duration
 	RetryInterval time.Duration
 }
 
 type cleanupFn func() error
 
-func (f *Framework) newTestCtx(t *testing.T) *TestCtx {
-	// TestCtx is used among others for namespace names where '/' is forbidden and must be 63 characters or less
+func (f *Framework) newContext(t *testing.T) *Context {
+
+	// Context is used among others for namespace names where '/' is forbidden and must be 63 characters or less
 	id := "osdk-e2e-" + uuid.New()
 
 	var namespace string
 	if f.singleNamespaceMode {
 		namespace = f.Namespace
 	}
-	return &TestCtx{
-		id:                 id,
-		t:                  t,
-		namespace:          namespace,
-		namespacedManPath:  *f.NamespacedManPath,
-		client:             f.Client,
-		kubeclient:         f.KubeClient,
-		restMapper:         f.restMapper,
-		skipCleanupOnError: f.skipCleanupOnError,
+	return &Context{
+		id:                id,
+		t:                 t,
+		namespace:         namespace,
+		namespacedManPath: *f.NamespacedManPath,
+		client:            f.Client,
+		kubeclient:        f.KubeClient,
+		restMapper:        f.restMapper,
 	}
 }
 
+// Deprecated: NewTestCtx exists for historical compatibility
+// You should use NewContext instead of.
+// todo(camilamacedo86): remove before 1.0.0
```

Comment:
```suggestion
```

Reply:
I think we should keep thee TODO in order to remember that it should be removed before 1.0.0

Reply:
This todo is reorganized [here](https://github.com/operator-framework/operator-sdk/pull/2549#discussion_r384785902).